packages/common/src/store/buy-usdc/sagas.ts

```diff
@@ -426,11 +435,50 @@ function* recoverPurchaseIfNecessary() {
         userBank: userBankAddress
       })
     )
-    yield* call(transferStep, {
-      wallet: rootAccount,
-      userBank,
-      amount
-    })
+
+    yield* call(
+      retry,
+      async () => {
+        const transferTransaction = await createRootWalletRecoveryTransaction(
+          audiusBackendInstance,
+          {
+            wallet: rootAccount,
+            userBank,
+            amount,
+            feePayer: feePayerKey
+          }
+        )
+        transferTransaction.partialSign(rootAccount)
+
+        console.debug(`Starting root wallet USDC recovery transaction...`)
+        const { res, error } = await relayTransaction(audiusBackendInstance, {
+          transaction: transferTransaction
+        })
+
+        if (res) {
+          console.debug(`Recovery transaction succeeded: ${res}`)
```

Comment:
super nit but i would throw if not res and return res if it succeeds

Reply:
We're already throwing in the 'not' case, just not using an else. I'm not sure if it makes sense to return the response here since the calling function won't need it or use it (The calling function being `watchRecovery` which is just watching for an action and then calling this function in response)